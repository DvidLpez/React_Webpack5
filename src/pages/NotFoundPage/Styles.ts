import styled from "styled-components";
import { Wrapper } from "../../components/StyledComponents/styledComponents";

export const WrapperNotFound = styled(Wrapper)`
  text-align: center;
`;
