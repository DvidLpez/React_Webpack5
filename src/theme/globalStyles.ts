
import { createGlobalStyle } from "styled-components";
import { theme } from "./theme";


export const GlobalStyle = createGlobalStyle`

   html {
      box-sizing: border-box;
   }
   body {
         background-color: ${theme.colors.tertiary};
         font-family: ${theme.fonts.style};
         padding: ${theme.spaces.none};
         margin: ${theme.spaces.none};
   }
   *,
   ::after,
   ::before {
      box-sizing: inherit;
   }

   a{
      text-decoration: none;
      color: ${theme.colors.primary};
   }

   hr{
      border: none;
      border-bottom: solid 1px ${theme.colors.secondary};
   }

   .icon_brand {
      font-size: ${theme.fonts.big};
      margin-right: ${theme.spaces.s15};
      color: ${theme.colors.title};
   }
   .icon_sweat {
      font-size: ${theme.fonts.extra};
   }

   .icon_error {
         font-size: ${theme.fonts.extraBig};
      }

   @media screen and (min-width: 768px) {
   
      .icon_brand {
         font-size: ${theme.fonts.extra};
      }
      
   }
`;
