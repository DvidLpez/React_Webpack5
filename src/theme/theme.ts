import { DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
  fonts: {
    style: "'Source Sans Pro', sans-serif",
    small: "0.9em",
    default: "1em",
    medium: "1.2em",
    large: "1.6em",
    big: "2em",
    extra: "3em",
    extraBig: "5em",
    bold: "600",
  },
  colors: {
    primary: "#000",
    secondary: "#dbdbdb",
    tertiary: "#fafafa",
    grey: "#414141",
    lightgrey: "#7c7c7c",
    gradient:
      "linear-gradient(to right, #3e9e7e 0%, #2e755d 10%, #205040 20%, #414141 30%)",
    title: "#fff",
    success: "#008000",
    closed: "#c60404",
    danger: "#932323",
    backComment: "#fff9d6",
    shadow: "2px 4px 11px 1px rgba(138, 138, 138, 1)",
  },
  spaces: {
    none: "0",
    center: "center",
    s5: "5px",
    s10: "10px",
    s15: "15px",
    s20: "20px",
    s25: "25px",
    s30: "30px",
    s50: "50px",
    p55: "5px 5px",
    p100: "10px 0",
    p1520: "15px 20px",
    p1010: "10px 10px",
    p1515: "15px 15px",
    p2020: "20px 20px",
    p2525: "25px 25px",
    p3030: "30px 30px",
    button: "10px 40px",
    borderRadius: "5px",
  }
};
