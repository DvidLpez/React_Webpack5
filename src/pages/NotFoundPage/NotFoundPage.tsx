import React, { FC, Fragment, ReactElement } from 'react';
import ButtomHome from '../../components/ButtomHome/ButtomHome';
import Header from '../../components/Header/Header';
import { Title, Subtitle } from '../../components/StyledComponents/styledComponents';
import { WrapperNotFound } from './Styles';


const NotFoundPage: FC = (): ReactElement => (
  <Fragment>
    <Header title="Page Not Found"/>
    <WrapperNotFound>
      <Title>404</Title>
      <Subtitle>PAGE NOT FOUND</Subtitle>
        <h4>
          THE PAGE YOU ARE LOOKING FOR DOES NOT EXIST.
          <br />
          PLEASE, SORRY FOR THE ANNOYANCE.
        </h4>
      <ButtomHome />
    </WrapperNotFound>
  </Fragment>
);
export default NotFoundPage;
