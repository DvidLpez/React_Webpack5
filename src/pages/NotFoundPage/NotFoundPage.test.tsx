import React from "react";
import { mount, shallow, render } from 'enzyme';
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme/theme";
import NotFoundPage from "./NotFoundPage";
import { MemoryRouter } from "react-router-dom";

export const mountWithTheme = (children: any) =>
   mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

export const renderWithTheme = (children: any) =>
   render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

export const shallowWithTheme = (children: any) =>
   shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>);



describe('Testing Page Not Found', () => {



   // it('renders snapshots, too', () => {
   //    const wrapper = shallow(<NotFoundPage />)
   //    expect(wrapper).toMatchSnapshot()
   // })



   it('renders snapshots, too', () => {
      const wrapper = mountWithTheme(
         <MemoryRouter initialEntries={['/random']}>
            <NotFoundPage />
         </MemoryRouter>
      )
      expect(wrapper).toMatchSnapshot()
   })

})