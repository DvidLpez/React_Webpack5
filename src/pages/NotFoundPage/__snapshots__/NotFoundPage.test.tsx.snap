// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Page Not Found renders snapshots, too 1`] = `
<Component
  theme={
    Object {
      "colors": Object {
        "backComment": "#fff9d6",
        "closed": "#c60404",
        "danger": "#932323",
        "gradient": "linear-gradient(to right, #3e9e7e 0%, #2e755d 10%, #205040 20%, #414141 30%)",
        "grey": "#414141",
        "lightgrey": "#7c7c7c",
        "primary": "#000",
        "secondary": "#dbdbdb",
        "shadow": "2px 4px 11px 1px rgba(138, 138, 138, 1)",
        "success": "#008000",
        "tertiary": "#fafafa",
        "title": "#fff",
      },
      "fonts": Object {
        "big": "2em",
        "bold": "600",
        "default": "1em",
        "extra": "3em",
        "extraBig": "5em",
        "large": "1.6em",
        "medium": "1.2em",
        "small": "0.9em",
        "style": "'Source Sans Pro', sans-serif",
      },
      "spaces": Object {
        "borderRadius": "5px",
        "button": "10px 40px",
        "center": "center",
        "none": "0",
        "p100": "10px 0",
        "p1010": "10px 10px",
        "p1515": "15px 15px",
        "p1520": "15px 20px",
        "p2020": "20px 20px",
        "p2525": "25px 25px",
        "p3030": "30px 30px",
        "p55": "5px 5px",
        "s10": "10px",
        "s15": "15px",
        "s20": "20px",
        "s25": "25px",
        "s30": "30px",
        "s5": "5px",
        "s50": "50px",
      },
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/random",
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "1tp33m",
              "pathname": "/random",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "1tp33m",
            "pathname": "/random",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NotFoundPage>
        <Header
          title="Page Not Found"
        >
          <Styled(styled.div)>
            <div
              className="sc-bdfBwQ sc-jrAGrp kUdDos fJoLyq"
            >
              <FontAwesomeIcon
                border={false}
                className="icon_brand"
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      496,
                      512,
                      Array [],
                      "f09b",
                      "M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z",
                    ],
                    "iconName": "github",
                    "prefix": "fab",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-github fa-w-16 icon_brand"
                  data-icon="github"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 496 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
              <Styled(styled.h1)>
                <h1
                  className="sc-gsTCUz sc-pFZIQ jrFgNj gURxYl"
                >
                  Page Not Found
                </h1>
              </Styled(styled.h1)>
            </div>
          </Styled(styled.div)>
        </Header>
        <Styled(styled.div)>
          <div
            className="sc-bdfBwQ sc-kEjbxe kUdDos rgIDo"
          >
            <styled.h1>
              <h1
                className="sc-gsTCUz jrFgNj"
              >
                404
              </h1>
            </styled.h1>
            <styled.h2>
              <h2
                className="sc-dlfnbm iVxDXV"
              >
                PAGE NOT FOUND
              </h2>
            </styled.h2>
            <h4>
              THE PAGE YOU ARE LOOKING FOR DOES NOT EXIST.
              <br />
              PLEASE, SORRY FOR THE ANNOYANCE.
            </h4>
            <ButtomHome>
              <Styled(styled.div)>
                <div
                  className="sc-bdfBwQ sc-fubCfw kUdDos fwLbUW"
                >
                  <Styled(Link)
                    to="/"
                  >
                    <Link
                      className="sc-gKsewC fMfLPa"
                      to="/"
                    >
                      <LinkAnchor
                        className="sc-gKsewC fMfLPa"
                        href="/"
                        navigate={[Function]}
                      >
                        <a
                          className="sc-gKsewC fMfLPa"
                          href="/"
                          onClick={[Function]}
                        >
                          Go Home
                        </a>
                      </LinkAnchor>
                    </Link>
                  </Styled(Link)>
                </div>
              </Styled(styled.div)>
            </ButtomHome>
          </div>
        </Styled(styled.div)>
      </NotFoundPage>
    </Router>
  </MemoryRouter>
</Component>
`;
